discovery/gossiper.go

```diff
@@ -2459,6 +2472,31 @@ func (d *AuthenticatedGossiper) handleChanAnnouncement(nMsg *networkMsg,
 		return nil, true
 	}
 
+	// Check if the channel is already closed in which case we can ignore
+	// it.
+	closed, err := d.cfg.ScidCloser.IsClosedScid(scid)
+	if err != nil {
+		log.Errorf("failed to check if scid %v is closed: %v", scid,
+			err)
+		nMsg.err <- err
+
+		return nil, false
+	}
+
+	if closed {
+		// Increment the peer's ban score if they are sending closed
+		// channel announcements.
+		if nMsg.isRemote {
+			d.banman.incrementBanScore(nMsg.peer.PubKey())
+		}
+
+		err = fmt.Errorf("ignoring closed channel %v", scid)
+		log.Error(err)
+		nMsg.err <- err
+
+		return nil, false
```

Comment:
In my testing I saw that the ban counter is increased, but we don't disconnect here, which should be the consequence if the ban counter reaches the threshold, right?

Reply:
fixed